ethclient/ethclient.go

```diff
@@ -70,7 +69,7 @@ var (
 )
 
 // Client defines interface for typed wrappers for the Ethereum RPC API.
-type Client interface {
+type Client interface { // Q: why do we need this?
```

Comment:
@felipemadero Can you remember why we needed this? I'd like to remove this interface (or move it elsewhere) to revert extra diffs introduced here.

Reply:
It's already removed in #882 😉 But we need to remove its import from avalanchego first I think (which is done in https://github.com/ava-labs/avalanchego/pull/3923)

Reply:
Ideally interfaces should be defined in the "accepting side" (so the cli) instead of being returned. See [accepting interfaces, returning concrete types](https://duncanleung.com/go-idiom-accept-interfaces-return-types/) 😉

Reply:
I think we can delete/move this interface in your PR. (but we should indeed remove it from here and reduce the diff in client code)